pimd/pim_iface.c

```diff
@@ -841,28 +832,42 @@ struct in_addr pim_find_primary_addr(struct interface *ifp)
 	}
 
 	for (ALL_LIST_ELEMENTS_RO(ifp->connected, node, ifc)) {
-		struct prefix *p = ifc->address;
+		pim_addr addr;
 
-		if (p->family != AF_INET) {
+		switch (ifc->address->family) {
+		case AF_INET:
+			v4_addrs++;
```

Comment:
This is now getting incremented for invalid ipv4 address (0.0.0.0) as well which differs from the previous behaviour. When IPv4 address is not configured, it tries to get the address from the loopback based on v4_addrs count outside of this loop. Please check.

Reply:
@mobash-rasool : I think, when IPv4 addr is not configured for an interface, then we will not have ifc->address->family set as AF_INET for that case. I think v4_addrs will not increment and it will remain 0 for IP fetch from loopback intf down below.

Reply:
The `0.0.0.0` address check there is bogus / cargo cult to begin with. It's probably carried over from some old code that used `ioctl` based kernel APIs where you get 0.0.0.0 if there is no IPv4 address configured. That is not the case with zebra, we get a list of addresses, and if there is no address the list is empty. There will never be a 0.0.0.0 address.

I've removed the check since it's bogus. The code overall is correct, even if it doesnt 100% match the previous code.

(btw: this is why this PR is separate from #10292 - in #10292 I tried to only do 100% identical changes.)